test/e2e/config/docker.yaml

```diff
@@ -253,7 +292,7 @@ providers:
     - sourcePath: "../data/infrastructure-docker/v1.3/cluster-template.yaml"
     - sourcePath: "../data/infrastructure-docker/v1.3/cluster-template-topology.yaml"
     - sourcePath: "../data/infrastructure-docker/v1.3/clusterclass-quick-start.yaml"
```

Comment:
please fix the paths to templates with v1.4 and add the new templates for v1.4 as in https://github.com/kubernetes-sigs/cluster-api/tree/main/test/e2e/data/infrastructure-docker/v1.3